wasmplugin/plugin.go

```diff
@@ -312,17 +350,19 @@ func (ctx *httpContext) OnHttpStreamDone() {
 	defer logTime("OnHttpStreamDone", currentTime())
 	tx := ctx.tx
 
-	// Responses without body won't call OnHttpResponseBody, but there are rules in the response body
-	// phase that still need to be executed. If they haven't been executed yet, now is the time.
-	if !ctx.processedResponseBody {
-		ctx.processedResponseBody = true
-		_, err := tx.ProcessResponseBody()
-		if err != nil {
-			proxywasm.LogCriticalf("failed to process response body: %v", err)
+	if !tx.IsRuleEngineOff() {
+		// Responses without body won't call OnHttpResponseBody, but there are rules in the response body
+		// phase that still need to be executed. If they haven't been executed yet, now is the time.
+		if !ctx.processedResponseBody {
+			ctx.processedResponseBody = true
+			_, err := tx.ProcessResponseBody()
+			if err != nil {
+				proxywasm.LogCriticalf("failed to process response body: %v", err)
+			}
 		}
+		ctx.tx.ProcessLogging()
```

Comment:
I had to skip `ProcessLogging` if ruleEngine is `Off` because of errors like:
```
  Invalid value for setvar "+tx.blocking_outbound_anomaly_score" on rule 980099
```
CRS at phase 5 looks for variables set at phase 1, but the latter does not happen when the engine is off.
Any thoughts? [Upstream Ref](https://github.com/corazawaf/coraza/blob/v3/dev/internal/corazawaf/transaction.go#L905)

Reply:
Shall we open an issue in coraza itself?

Reply:
Yup, better have a broader discussion. Opened https://github.com/corazawaf/coraza/issues/520 and added a TODO in this PR